### PR TITLE
Inplace

### PR DESCRIPTION
**Pull request by jjpr-mit**

### CODE REVIEW

File: brainio_base/assemblies.py [hunk 1]
```diff
@@ -20,11 +20,17 @@ class DataAssembly(DataArray):
     """A DataAssembly represents a set of data a researcher wishes to work with for
     an analysis or benchmarking task.  """
 
+    __slots__ = ()
+
     def __init__(self, *args, **kwargs):
-        super(DataAssembly, self).__init__(*args, **kwargs)
-        gather_indexes(self)
+        temp = DataArray.__new__(DataArray, *args, **kwargs)
+        DataArray.__init__(temp, *args, **kwargs)
```
Comment (mschrimpf) [inline brainio_base/assemblies.py]: is this really necessary? Doesn't `temp = DataArray(*args, **kwargs)` result in the same thing?

File: brainio_base/assemblies.py [hunk 2]
```diff
@@ -20,11 +20,17 @@ class DataAssembly(DataArray):
...
+        temp = gather_indexes(temp)
```
Comment (mschrimpf) [inline brainio_base/assemblies.py:27]: a comment to explain what's going on / why it's necessary would be useful for future maintenance

File: brainio_base/assemblies.py [hunk 3]
```diff
@@ -172,29 +178,29 @@ def multisel(self, method=None, tolerance=None, drop=False, **indexers):
 
 class BehavioralAssembly(DataAssembly):
     """A BehavioralAssembly is a DataAssembly containing behavioral data.  """
-    pass
+    __slots__ = ()
```
Comment (mschrimpf) [inline brainio_base/assemblies.py:189]: why is this needed?

File: tests/test_assemblies.py [hunk 4]
```diff
@@ -49,14 +41,28 @@ def test_int_multi_coord(self):
     @pytest.mark.skip(reason="not implemented")
     def test_multi_dim(self):
         d = DataAssembly([[1, 2, 3], [4, 5, 6], [7, 8, 9], [10, 11, 12]],
-                         coords={'a': ['a', 'a', 'b', 'b'],
+                         coords={'a': ['alpha', 'alpha', 'beta', 'beta'],
```
Comment (mschrimpf) [inline tests/test_assemblies.py]: if these change, the ones below (`assert g.equals(...)`) should also change no? (this test is skipped)

File: tests/test_assemblies.py [hunk 5]
```diff
@@ -5,24 +5,16 @@
...
         d = DataAssembly([[1, 2, 3], [4, 5, 6]],
                          coords={'a': ('multi_dim', ['a', 'b']), 'b': ('multi_dim', ['c', 'c']), 'c': ['x', 'y', 'z']},
                          dims=['multi_dim', 'c'])
-        g = d.multi_groupby(['a']).mean()
-        assert g.equals(DataAssembly([2, 5], coords={'multi_dim': ['a', 'b']}, dims=['multi_dim']))
-        # ideally, we would want `g.equals(DataAssembly([2, 5],
-        #   coords={'a': ('multi_dim', ['a', 'b']), 'b': ('multi_dim', ['c', 'c'])}, dims=['multi_dim']))`
-        # but this is fine for now.
+        with pytest.raises(ValueError):
+            g = d.multi_groupby(['a']).mean()
```
Comment (mschrimpf) [inline tests/test_assemblies.py]: this changes tests to expect failure where they passed before --> hand off single-grouping to regular `groupby` if it occurs

**Review by mschrimpf — COMMENTED** [review]
Comments would be helpful to explain why certain things are necessary (e.g. `__new__` and `__slots__`).
Also it looks like `multi_groupby` tests have been changed to expect a ValueError instead of passing?

File: tests/test_assemblies.py (hunk 4 above)
Comment (mschrimpf) [reply tests/test_assemblies.py]: let's just delete this test, it's dead code at this point since it's skipped

Comment (jjpr-mit) [reply tests/test_assemblies.py]: Reverted.  I tried making a function to produce DataArrays with more typical coordinates.  I'll remove it if it's gross.

**Review by mschrimpf — APPROVED** [review]
